src/data/nat/totient.lean

```diff
@@ -194,6 +205,16 @@ begin
   rwa [succ_le_iff, pos_iff_ne_zero],
 end
 
+lemma card_units_zmod_lt_sub_one {p : ℕ} (hp : 1 < p)
+  (hp' : fact (0 < p) := ⟨zero_lt_one.trans hp⟩) :
+  fintype.card (units (zmod p)) ≤ p - 1 :=
```

Comment:
This assumption seems pretty weird to me; Perhaps it would be better to write
```suggestion
lemma card_units_zmod_lt_sub_one {p : ℕ} (hp : 1 < p) [fintype (units (zmod p))] :
  fintype.card (units (zmod p)) ≤ p - 1 :=
```

Reply:
@jcommelin was the one who wrote it this way, maybe he has an opinion. My understanding was that this way would be good, because you don't have to write this typeclass argument when you use it.

Reply:
@BoltonBailey Probably the suggestion by Eric is better. If `fintype` didn't contain data, all of this would be unproblematic.